charts/k8s-service/templates/canarydeployment.yaml

```diff
@@ -0,0 +1,325 @@
+{{- /*
+The Canary Deployment Controller for the application being deployed. This resource manages the creation and replacement
+of only the canary Pod(s) backing your application. It is intended to be used to test new release candidates
+and ensure they are free of issues prior to performing a full roll out.
+*/ -}}
+{{- /*
+We must decide whether or not there are volumes to inject. The logic to decide whether or not to inject is based on
+whether or not there are configMaps OR secrets that are specified as volume mounts (`as: volume` attributes). We do this
+by using a map to track whether or not we have seen a volume type. We have to use a map because we can't update a
+variable in helm chart templates.
+
+Similarly, we need to decide whether or not there are environment variables to add
+
+We need this because certain sections are omitted if there are no volumes or environment variables to add.
+*/ -}}
+
+{{/* Go Templates do not support variable updating, so we simulate it using dictionaries */}}
+{{- if .Values.canary.enabled -}}
```

Comment:
NIT: I'd put this at the very top, right after the preamble docs. Putting this here feels a bit out of place given all the docs on volumes.

Reply:
Fixed in 9428a46